### DISCUSSION TITLE
How to add new field arguments on existing schema? (SchemaTransformer?)

### DISCUSSION THREAD

**Discussion by yarc** [graphql-java/graphql-java#2618] [Q&A]
Hi,

I have a GraphQLSchema on which i need to add some new field arguments. I've been trying to use the SchemaTransformer and visitGraphQLFieldDefinition to modify the field definition and then call context.changeNode(transformed), but the new arguments are not present in the transformed schema.

Am i misusing the transformer? What would be the easiest way to add arguments to existing types?

Many thanks

Edit: I'm more or less stuck with version 16.2

**Comment by andimarek** [comment]
Can you please post a minimal example to reproduce the problem? Your idea sounds about right, but without actual code it is not really possible to say what goes wrong.
